### ISSUE TITLE
Issue starting KBFS windows app following registration of mobile device

### ISSUE THREAD

**Issue by jtsmith0101** [keybase/client#6755]
my workflow:
- saw Marco set me up on Google Play
- Installed Keybase app onto phone
- logged into my account via keybase login (PGP)
- named my device Galaxy_S8
- device appeared to freeze; set it down for a bit in case it was syncing
- looked at my conversations, I had a new one from user marcopolo; unable to read
- viewed my folders; noticed a "rekey" tag on almost all of them.
- clicked a folder; greeted with message I needed to log in from windows application or risk losing files.
- Stopped work from the phone; started up laptop
- Keybase unable to login on Laptop

my log id: ef91f102cfe877cd9b1ea11c

**Comment by jtsmith0101** [comment]
I appear able to login from the command line, however the app is still "hung" on "loading"

**Comment by jtsmith0101** [comment]
My devices are displayed as they should be from the command line. (keybase device list)

**Comment by jtsmith0101** [comment]
could be a coincidence?

**Comment by jtsmith0101** [comment]
app starting to "wake up"...

See log:  fdb733aaa445633b8eac9c1c

There is an error in my log sending:  
- INFO ignoring error getting keybase status: Timed out waiting for pipe '\\.\pipe\kbservice\Users\\AppData\Local\Keybase\keybased.sock' to come available
- INFO ignoring UI logs: Timed out waiting for pipe '\\.\pipe\kbservice\Users\\AppData\Local\Keybase\keybased.sock' to come available

The task bar widget displays useful messages, such as folder rekeys, when I open the App, it displays that the app is "Disconnected"

**Comment by jtsmith0101** [comment]
hard restart of computer..... still exhibiting same symptoms.

See log:  4152d40fdd3afa8a36f7c41c

**Comment by jtsmith0101** [comment]
Output from logout command:
C:\Users\>keybase logout
- INFO Starting background server with pid=9384
- WARNING Run error: RPC error. type: 0, method: keybase.1.apiserver.Get, length: 4, error: error decoding message field at position 3, error: [pos 29]: readContainerLen: Unrecognized descriptor byte: hex: 83, decimal: 131
- WARNING (CONN keybase service 86daa94a) Connection: error calling OnConnect handler: error running SyncAll: context canceled
- INFO RemoteSettingsRepairman moving from C:\Users\jtsmi\AppData\Roaming\Keybase to C:\Users\\AppData\Local\Keybase

**Comment by MarcoPolo** [comment]
cc @zanderz

**Comment by cjb** [comment]
CC @maxtaco @patrickxb @zanderz

**Comment by maxtaco** [comment]
Looks like a bug in the routine that moved home dirs on windows after the
reboot.

Not sure why the original problem started.

**Comment by maxtaco** [comment]
Sorry @jtsmith0101 for your very bad experience. This is highly embarrassing :(

We'll investigate as soon as we can. Thanks for your patience.

**Comment by jtsmith0101** [comment]
Happy to test

**Comment by zanderz** [comment]
The RemoteSettingsRepairman warning is a red herring, since fixed. Updater log shows the target install directory to have been %LOCALAPPDATA%\Keybase since mid December.

With the RPC errors, I am curious whether a mix of versions may be involved? When you run it from c:\Users, which one runs (as per `which keybase`)?

The installer logs are not present. Maybe your %TEMP% directory got cleaned out?

Does it come back if you reboot?

**Comment by jtsmith0101** [comment]
No change I'm afraid.

I do remember the change in installation folders from Appdata\Roaming to
Appdata\Local as I had to manually change my path.

I do seem to have fully populated keybase folders in both locations.
Perhaps the old pre-December installation wasn't removed when the
post-December / current version installed?

I wouldn't expect this to affect performance.  Before yesterday, Keybase
behaved reliably.

Maybe uninstall, download the fresh version and reinstall?

**Comment by zanderz** [comment]
What do you get when you type `which keybase` ?

**Comment by jtsmith0101** [comment]
'which' doesn't appear to be a valid Windows command.... ;)

**Comment by maxtaco** [comment]
Mixed versions of the app would not account for that error. We would only
see that error if the client started sending non msgpack data down the
pipe, or maybe if the pipe became unreliable and started dropping data. It
is really hard to know.

It looks like the service is running in auto fork mode.  Is that expected?
That doesn't seem right on windows.

**Comment by zanderz** [comment]
Whoops, I meant `where keybase` - but that probably is not very helpful, as per Max.

**Comment by jtsmith0101** [comment]
fair enough.

It displays \AppData\Local\Keybase

**Comment by jtsmith0101** [comment]
I'm beginning to think something corrupted while I was adding my phone to my devices...

**Comment by jtsmith0101** [comment]
logged in from the command prompt.  Greeted with a new popup window for entering my passphrase.  Logged in.... 

The taskbar widget, when clicked looks normal.  See attached.
![image](https://cloud.githubusercontent.com/assets/10441596/25508827/091d7a34-2bf8-11e7-823b-8fd2689ba377.png)

When I select "Open Keybase" from the drop down menu, the application appears stuck here:
![image](https://cloud.githubusercontent.com/assets/10441596/25508875/3c48bf86-2bf8-11e7-946f-d8790600a8ce.png)

**Comment by maxtaco** [comment]
@strib  / @akalin there might be a KBFS problem in this dump, the first one linked: fdb733aaa445633b8eac9c1c

```
172745 2017-04-26T16:58:21.726300+10:00 - [WARN kbfs journal_server.go:229] ee93 Couldn't enable journal for 026bde44c99940882e6cf65534ab6816: Current UID is empty
172746 github.com/keybase/kbfs/libkbfs.(*JournalServer).enableLocked
172747 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/journal_server.go:433
172748 github.com/keybase/kbfs/libkbfs.(*JournalServer).Enable
172749 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/journal_server.go:485
172750 github.com/keybase/kbfs/libkbfs.(*JournalServer).getTLFJournal
172751 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/journal_server.go:227
172752 github.com/keybase/kbfs/libkbfs.(*JournalServer).JournalStatus
172753 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/journal_server.go:726
172754 github.com/keybase/kbfs/libkbfs.(*folderBranchOps).getJournalPredecessorRevision
172755 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/folder_branch_ops.go:613
172756 github.com/keybase/kbfs/libkbfs.(*folderBranchOps).getMostRecentFullyMergedMD
172757 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/folder_branch_ops.go:1110
172758 github.com/keybase/kbfs/libkbfs.(*folderBlockManager).doReclamation
172759 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/folder_block_manager.go:1034
172760 github.com/keybase/kbfs/libkbfs.(*folderBlockManager).reclaimQuotaInBackground
172761 C:/Jenkins/workspace/windows-installer_0/src/github.com/keybase/kbfs/libkbfs/folder_block_manager.go:1166
172762 runtime.goexit
172763 C:/Go/src/runtime/asm_386.s:1612
172764 2017-04-26T16:58:22.933255+10:00 - [DEBU kbfs fs.go:565] ee94 => FS CreateFile [tags:DID=XvdNmhOWjFMS-JkFz1vkyA]
```

**Comment by strib** [comment]
@maxtaco: that only means KBFS thinks the user isn't logged in.  It's normal for logged out users.  Sorry it looks scary.

**Comment by maxtaco** [comment]
K thanks, sorry for the false alaram.

**Comment by maxtaco** [comment]
@jtsmith0101 are you running any sort of virus protection?

**Comment by jtsmith0101** [comment]
yes.  I've given explicit permission to the following apps:
kbfsdokan
Keybase
Keybase utility
upd

All are parked within \AppData\Local

**Comment by maxtaco** [comment]
Did you recently change anything here?  The issue, at its core, is that the different components of the system can't communicate with each other.

**Comment by maxtaco** [comment]
In particular, the `keybase` process runs as a background service and listens for connections on this pipe:

```
pipe:\\.\pipe\kbservice\Users\jtsmi\AppData\Local\Keybase\keybased.sock
```
On background servers are failing to connect to it.  Is this socket firewalled off by your virus protector?

**Comment by jtsmith0101** [comment]
I've not changed anything.  Possible poor timing with Norton360 .... should
I add another fw exclusion for an unamed application / service?

**Comment by maxtaco** [comment]
Can you try to disable norton for a moment and then try again?

**Comment by zanderz** [comment]
We have made a fix, which unfortunately is merely speculative, since we can't reproduce the problem. We would appreciate it if you would try installing https://s3.amazonaws.com/prerelease.keybase.io/windows/Keybase_1.0.21-20170428134902%2B4280d08.386.exe

Thanks.

**Comment by jtsmith0101** [comment]
This had no effect.

**Comment by jtsmith0101** [comment]
My Keybase appears back to normal.

Thanks guys!

**Comment by zanderz** [comment]
Good news! What fixed it?

**Comment by jtsmith0101** [comment]
new installation file

**Comment by jtsmith0101** [comment]
or pure vodoo magic

**Comment by zanderz** [comment]
Thanks for your patience and help.

**Comment by jtsmith0101** [comment]
happy to test ;)

**Comment by jtsmith0101** [comment]
And most importantly, thanks for being so responsive

**Comment by jtsmith0101** [comment]
I think this can be closed.  Thanks for all your help.  I spent some time digging in my AV (following a rummage through the outstanding issues on GH) and discovered that runquiet.exe is being flagged as a trojan and quarantined.  After removing the quarantine, restoring the file and rebooting, keybase behaved as I expected.  I've whitelisted all of the contents of the \keybase folder... so AV 'shouldn't' impact performance again.
